sklearn/neighbors/_base.py

```diff
@@ -1021,6 +1021,16 @@ class from an array representing our data set and ask who's
                 neigh_ind_list = sum(chunked_results, [])
                 results = _to_object_array(neigh_ind_list)
 
+            if sort_results:
+                if not return_distance:
+                    raise ValueError("return_distance must be True "
+                                     "if sort_results is True.")
```

Comment:
Should this validation be done in all cases, and not just with 'brute'?

Reply:
This validation is already done in the method `query_radius` for "ball_tree" and "kd_tree", and it is not necessary with a precomputed sparse graph.